File: Server/src/Server.Application/ViewModels/AdminUserViewModel.cs

namespace Server.Application.ViewModels;

public class AdminUserViewModel
{
    public Guid Id { get; set; }
    public string Name { get; set;}
    public string AvatarImageUrl { get; set; }
    public string Email { get; set;}
    public bool IsCritic { get; set; }
}

File: client/src/models/user/AdminUserViewModel.ts
import { z } from 'zod';

const AdminUserViewModel = z.object({
  id: z.string(),
  name: z.string(),
  avatarImageUrl: z.string().nullable(),
  email: z.string().email(),
  isCritic: z.boolean()
});

export default AdminUserViewModel;
export type AdminUserViewModelType = z.infer<typeof AdminUserViewModel>;
